lib/reservation.rb

```diff
@@ -0,0 +1,23 @@
+require "date"
+
+module HotelSystem
+  class Reservation
+    attr_reader :room, :start_date, :end_date, :price, :guest, :reservation_id
+    attr_accessor :cost
+
+    def initialize(room:, start_date:, end_date:, guest:)
+      @start_date = start_date
+      @end_date = end_date
+      @room = room
+      @price = 200
+      @cost = calculate_cost
+      @guest = guest
+      @reservation_id = ((0...5).map { rand(10) }).join.to_s
+      raise ArgumentError, "Invalid date range" if @end_date <= @start_date
+    end
+
+    def calculate_cost
+      return ("%.2f" % ((@end_date - @start_date) * @price)).to_f
+    end
```

Comment:
I like the decision to make this a method rather than storing the cost as an instance variable. That way if the number of nights or the cost per night were to change (say in a future version of the program), the total cost will automatically be correct.